examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php

```diff
@@ -114,54 +121,45 @@ public function showReportAction(
         // Determines the number of the page to load (the first one by default).
         $pageNo = $request->input('page') ?: 1;
 
+        // Page number of the Google Ads API result is not the same as the requested UI page number.
+        // This is because Google Ads API has a defined default page size and users cannot specify
+        // it.
+        $resultPageNo = intval($entriesPerPage * $pageNo / self::DEFAULT_PAGE_SIZE);
```

Comment:
Line 142 should cope with the last page token that you mentioned.
We've tried to optimize things here since the last version by caching the page tokens while sending a request to the API. 
If we're going to change the condition on line 130 to do what you mentioned, that's possible, but would we lose the optimization here?

Reply:
I see what you mean about the last page token, so that's fine as-is.
Instead of relying on the default page size, though, what if instead you kept track of the size of each page for each page token? You could do this around line 142 based on the results count for the page just retrieved.

Reply:
Not sure I get what you mean. The page size shouldn't be different for each request I believe.
And now the Google Ads API always set the page size for search() at 10,000, so there is no variation here.

Reply:
`$resultPageNo` is necessary for the if condition at line 131 and it happens before the first request.
And I don't think there is a good way to get the page size before calculating $resultPageNo at line 128 unless we send one request first. 
This would result in more complicated logic here and I doubt we can say that it's more efficient (because we would spend one request for 10000 rows for a large set of results).

Do I miss anything here?

Reply:
I see what you're saying. I took another look and I agree it's going to get _very_ complicated if we worry (probably unnecessarily) about varying page sizes.